xm-commons-logging-configurable/src/main/java/com/icthh/xm/commons/logging/configurable/LoggingRefreshableConfiguration.java

```diff
@@ -0,0 +1,136 @@
+            LoggingConfig spec = ymlMapper.readValue(config, LoggingConfig.class);
+
+            if (spec == null) {
+                this.apiLoggingConfig.remove(tenant);
+                this.serviceLoggingConfig.remove(tenant);
+                this.lepLoggingConfig.remove(tenant);
+                return;
+            }
+
+            this.serviceLoggingConfig.put(tenant, spec.buildServiceLoggingConfigs());
+            this.apiLoggingConfig.put(tenant, spec.buildApiLoggingConfigs());
+            this.lepLoggingConfig.put(tenant, spec.buildLepLoggingConfigs(tenant));
+
+            log.info("Tenant configuration was updated for tenant [{}] by key [{}]", tenant, updatedKey);
+        } catch (Exception e) {
+            log.error("Error read tenant configuration from path " + updatedKey, e);
+        }
+    }
+
+    @Override
+    public boolean isListeningConfiguration(final String updatedKey) {
+        return this.matcher.match(mappingPath, updatedKey);
+    }
+
+    @Override
+    public void onInit(final String configKey, final String configValue) {
+        if (this.isListeningConfiguration(configKey)) {
+            this.onRefresh(configKey, configValue);
+        }
+    }
+
+    @Override
+    public LogConfiguration getServiceLoggingConfig(String packageName,
+                                                              String className,
+                                                              String methodName) {
+        return getLogConfiguration(serviceLoggingConfig.get(getRequiredTenantKeyValue(this.tenantContextHolder)),
+            packageName,
+            className,
+            methodName);
+
+    }
+
+    @Override
+    public LogConfiguration getApiLoggingConfig(String packageName, String className, String methodName) {
+        return getLogConfiguration(apiLoggingConfig.get(getRequiredTenantKeyValue(this.tenantContextHolder)),
+            packageName,
+            className,
+            methodName);
+    }
+
+    @Override
+    public LepLogConfiguration getLepLoggingConfig(String fileName) {
+        String tenantKey = getRequiredTenantKeyValue(this.tenantContextHolder);
+
+
+        Map<String, LepLogConfiguration> logConfiguration = lepLoggingConfig.get(tenantKey);
+        if (MapUtils.isEmpty(logConfiguration)) {
+            return null;
+        }
+        if (logConfiguration.containsKey(fileName)) {
+            return logConfiguration.get(fileName);
+        }
+        return null;
+    }
+
+
+    private LogConfiguration getLogConfiguration(Map<String, LogConfiguration> logConfiguration,
+                                                           String packageName,
+                                                           String className,
+                                                           String methodName) {
+
+        if (MapUtils.isEmpty(logConfiguration)) {
+            return null;
+        }
+
+        if (logConfiguration.containsKey(className)) {
```

Comment:
i agree with @amedvedchuk 
replace containsKey to null check

Reply:
changed